### ISSUE TITLE
False positive Warning.REFERENCE_EQUALITY for class with Interface field if Interface defines equals()

### ISSUE THREAD

**Issue by tazle** [jqno/equalsverifier#130]
Given following classes:

Interface that defines equals():

``` java
public interface Interface {
    long getValue();
    boolean equals(Object other);
}
```

A class using it: (standard IDEA boilerplate)

``` java
public final class Class {
    private final Interface f1;

    public Class(Interface f1) {
        this.f1 = f1;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Class aClass = (Class) o;

        return !(f1 != null ? !f1.equals(aClass.f1) : aClass.f1 != null);

    }

    @Override
    public int hashCode() {
        return f1 != null ? f1.hashCode() : 0;
    }
}
```

the following test fails with Warning.REFERENCE_EQUALITY:

``` java
import nl.jqno.equalsverifier.EqualsVerifier;
import org.junit.Test;


public class ClassTest {

    @Test
    public void testEq() {
        EqualsVerifier.forClass(Class.class).verify();
    }
}
```

Maybe this is related to #113 and #104?

**Comment by jqno** [comment]
Hi,

Thanks for reporting this. I've reproduced this, and I agree it's a bug. I'll fix it in the next release.

**Comment by jqno** [comment]
I've just released version 1.7.6, which fixes this.
